app/models/host/managed.rb

```diff
@@ -658,13 +658,13 @@ def provider
   # no need to store anything in the db if the password is our default
   def root_pass
     return self[:root_pass] if self[:root_pass].present?
-    return hostgroup.try(:root_pass) if hostgroup.try(:root_pass).present?
-    Setting[:root_pass]
+    return hostgroup.try(:root_pass) if hostgroup.try(:root_pass)
```

Comment:
Why did we removed `present?` here?

Reply:
I removed it because there was a `present?` and  ignored the method where is used. On the second look, it should be there.